scalanativelib/src/mill/scalanativelib/testinterface/ComRunner.scala

```diff
@@ -0,0 +1,128 @@
+package mill.scalanativelib
+package testinterface
+
+import java.io._
+import java.net.{ServerSocket, SocketTimeoutException}
+
+import mill.modules.Jvm
+
+import scala.concurrent.TimeoutException
+import scala.concurrent.duration.Duration
+
+//import sbt.{Logger, MessageOnlyException}
+
+import scala.scalanative.testinterface.serialization.Log.Level
+import scala.scalanative.testinterface.serialization._
+//import scala.scalanative.sbtplugin.SBTCompat._
+
+case class MessageOnlyException(msg: String) extends Exception(msg)
+
+/**
+ * Represents a distant program with whom we communicate over the network.
+ * @param bin    The program to run
+ * @param args   Arguments to pass to the program
+ * @param logger Logger to log to.
+ */
+class ComRunner(bin: File,
```

Comment:
We're going to refactor test runner out of sbt plugin in `0.3.8` (see https://github.com/scala-native/scala-native/pull/1234). It was an unfortunate oversight not to do it sooner.

Reply:
@densh Thanks for the review